src/cpp/src/device_config.hpp

```diff
@@ -80,11 +80,10 @@ class DeviceConfig {
             OPENVINO_THROW(m_device, " is not supported by OpenVINO Continuous Batching");
         }
 
-        OPENVINO_ASSERT(scheduling_config.num_kv_blocks > 0 || scheduling_config.cache_size > 0, "num_kv_blocks or cache_size should be more than zero.");
         if (scheduling_config.num_kv_blocks > 0) {
             m_num_kv_blocks = scheduling_config.num_kv_blocks;
         }
-        else {
+        else if (scheduling_config.cache_size > 0) {
```

Comment:
let's also drop scheduler config usage in places like here
https://github.com/openvinotoolkit/openvino.genai/blob/f177ffc9799ef34a57b257e1811a60c68c167eb2/samples/cpp/speculative_decoding_lm/speculative_decoding_lm.cpp#L29-L30
as it's not required anymore